packages/client/src/views/RegisterForm/RegisterForm.tsx

```diff
@@ -354,7 +369,21 @@ class RegisterFormView extends React.Component<FullProps, State> {
   }
 
   onSaveAsDraftClicked = async () => {
-    await this.props.writeApplication(this.props.application)
+    const { application } = this.props
+    const isRegistrarOrRegistrationAgent =
+      this.userHasRegisterScope() || this.userHasValidateScope()
+    const isApplicationDeclaredOrRejected =
+      application.registrationStatus === DECLARED ||
+      application.registrationStatus === REJECTED
```

Comment:
I think, we need to add `VALIDATED` state check here as well. Because registrar can make further modification even after the application is already validated by a registration agent.

Reply:
Thanks, @mushrafulhoque-dsi agreed. I am on it